dlls/winex11.drv/mouse.c

```diff
@@ -547,7 +547,7 @@ BOOL clip_fullscreen_window( HWND hwnd, BOOL reset )
     /* maximized windows don't count as full screen */
     if ((style & WS_MAXIMIZE) && (style & WS_CAPTION) == WS_CAPTION) return FALSE;
     if (!(data = get_win_data( hwnd ))) return FALSE;
-    fullscreen = is_window_rect_fullscreen( &data->whole_rect );
+    fullscreen = fs_hack_enabled() || is_window_rect_fullscreen( &data->whole_rect );
```

Comment:
Did I made correct assumption that fs_hack_enabled is true only when in the fullscreen mode?

Reply:
This looks wrong to me. `is_window_rect_fullscreen` should already handle the fullscreen hack via the primary monitor getting resized in `xinerama_init`. `fs_hack_enabled` only indicates that we are emulating a mode change for the application, not that there is a fullscreen window present.

Reply:
> fs_hack_enabled only indicates that we are emulating a mode change for the application, not that there is a fullscreen window present.

Yeah... the change is probably wrong. It attempts to fixes two issues on my machine:
1. I don't remember exactly, but I think when the game is started and the fs hack is enabled, data->whole_rect is a rect that has the same size as the primary screen but slightly moved to one side. I will look into it again tomorrow and give more info.
2. When changing resolution, data->whole_rect may have the size same as the old resolution and not the new one.

I have tested it with one monitor. I have a tv connected to the video card but it was disabled in the settings. It may have interfered. I also will look into it.

Reply:
Come to think of it, the issues probably do occur elsewhere. clip_fullscreen_window() seems to be called a bit too soon before the window size is updated.